pym/portage/meta-manifest.py

```diff
@@ -0,0 +1,276 @@
+		myrepo = path.split(os.path.sep)
+		repolevel = len(myrepo)
+		return repolevel
+
+	def find_repotype(self, path):
+		'''Finds the type of the repo directory'''
+		if len(set(['metadata', 'eclass', 'profiles']).intersection(os.listdir(path))) > 0:
+			return 'root'
+		elif path.endswith("eclass/"):
+			return 'eclass'
+		elif path.endswith("profiles/"):
+			return 'profile'
+		else:
+			if self.find_repolevel(path) == 2:
+				return 'category'
+			else:
+				return 'package'
+
+	def create(self):
+		'''Creates a MetaManifest file'''
+		repodir = self.pkgdir
+		repotype = self.find_repotype(repodir)
+		if repotype == 'profile':
+			self.create_profile()
+		elif repotype == 'eclass':
+			self.create_eclass()
+		elif repotype == 'root':
+			self.create_master()
+		else:
+			self.create_cat()
+
+	def create_master(self):
+		'''Creates a MetaManifest file from all the MetaManifest files of the repository'''
+		self.find_reporoot()
+		myrepodir = self.reporoot
+		self.pkgdir = self.reporoot
+		self.fhashdict = {}
+		for ftype in MANIFEST2_IDENTIFIERS:
+			self.fhashdict[ftype] = {}
+		for file in os.listdir(myrepodir):
+			fpath = os.path.join(myrepodir, file)
+			if os.path.isfile(fpath):
+				if not fpath.endswith('Manifest'):
+					ftype = guessManifestFileType(fpath, is_pkg=False)
+					self.fhashdict[ftype][f] = perform_multiple_checksums(fpath, self.hashes)
+			else:
+				for mf in os.listdir(fpath):
+					mfpath = os.path.join(fpath, mf)
+					if mfpath.endswith('Manifest'):
+						f = mfpath.replace(self.reporoot, "")
+						self.fhashdict["MANIFEST"][f] = perform_multiple_checksums(mfpath, self.hashes) 
+
+	def create_eclass(self):
+		'''Creates a MetaManifest file in the eclass directory'''
+		eclass_dir = self.pkgdir
+		self.fhashdict = {}
+		for ftype in MANIFEST2_IDENTIFIERS:
+			self.fhashdict[ftype] = {}
+
+		for eclass_dir, eclassdir_dir, files in os.walk(eclass_dir):
+			for f in files:
+				try:
+                                        f = _unicode_decode(f,encoding=_encodings['fs'], errors='strict')
+                                        eclass_dir = _unicode_decode(eclass_dir,encoding=_encodings['fs'], errors='strict')
+				except UnicodeDecodeError:
+                        	        continue
+				fpath = os.path.join(eclass_dir, f)
+				ftype = guessManifestFileType(fpath, is_pkg=False)
+				f = fpath.replace(self.pkgdir, "")
+				if not f.endswith("Manifest"):
+					self.fhashdict[ftype][f] = perform_multiple_checksums(fpath, self.hashes)
+
+	def create_cat(self):
+		'''Creates a MetaManifest file in the selected category'''
+		catdir = self.pkgdir
+		for ftype in MANIFEST2_IDENTIFIERS:
+                        self.fhashdict[ftype] = {}
+		for pkg_dir in os.listdir(catdir):
+			pkg_dir = os.path.join(catdir, pkg_dir)
+			if os.path.isdir(pkg_dir):
+				for file in os.listdir(pkg_dir):
+					fpath = os.path.join(pkg_dir, file)
+					if os.path.isfile(fpath) and fpath.endswith('Manifest'):
```

Comment:
Another abuse of `endswith`. Alternative:
```
if file == 'Manifest' and os.path.isfile(fpath):
```
Do the string comparison first, in order to skip the stat call if the string doesn't match.